scripts/makeLegacyPlots.py

```diff
@@ -425,10 +425,26 @@ def retrieveShapes (rootFile, namelist, var, sel, reg, shapeNameList):
         # because the QCD syst is only lnN and no shape variations are produced
         if 'QCD' in name: continue
         for shapeName in shapeNameList:
-            fullName = name + "_" + sel + "_" + reg + "_" + var + '_' + shapeName
+
+            # Possible to make plots both for DNN output...
+            if 'DNN' in var:
+                fullName = name + "_" + sel + "_" + reg + "_" + var + '_' + shapeName
+            # ...and for other varaibles...
+            else:
+                # ...that do not have the ES shifted variations...
+                if 'tauup' in shapeName or 'taudown' in shapeName or \
+                   'eleup' in shapeName or 'eledown' in shapeName or \
+                   'muup'  in shapeName or 'mudown'  in shapeName or \
+                   'jetup' in shapeName or 'jetdown' in shapeName:
+
+                    fullName = name + "_" + sel + "_" + reg + "_" + var
+
+                # ...but still have the other shapes (PUjetID, tauID, trigSF...)
+                else:
+                    fullName = name + "_" + sel + "_" + reg + "_" + var + '_' + shapeName
```

Comment:
I am not sure I see the point of doing this: in this way you are skipping all the ES shapes, but this can just be done by calling `--removeESsystBand` as an input to the call of the `makeLegacyplots.py` script.

From what I understand here you always skip the ES and retrieve the central value. This will give a contribution of 0 to the syst band. But this is a waste of time when plotting. For variables without ES you just call `--removeESsystBand` and the plotting time will be reduced compared to this approach.

So if I have understood well what you are doing here, this is not needed at all. I would just leave as it was and just not retrieve at all the ES shape when it is not produced.